OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -264,7 +265,7 @@ public IEnumerable<PipType> GetPips(Actor self)
 
 		PipType GetPipAt(int i)
 		{
-			var n = i * Info.MaxWeight / Info.PipCount;
+			var n = Info.PipCount >= 0 ? i * Info.MaxWeight / Info.PipCount : i / -Info.PipCount;
```

Comment:
This behaves counter-intuitively ingame.  If you have a half-full transport with `PipCount: -2` then I would expect that the pip row would be half full, not full.